### PR TITLE
feat: support config AddressResolverGroup in r2dbc-mysql

### PR DESCRIPTION
**Pull request by hezhangjian**
Motivation:
Currently,`AddressResolverGroup` can't be configured. The DnsResolver default start address listen to "0.0.0.0", which may have some security risks.
also see https://github.com/netty/netty/pull/11061

Modification:
Add `AddressResolverGroup` in Client's connect method

### CODE REVIEW

**Review by CennyMo — APPROVED** [review]

**Comment by hezhangjian** [comment]
@jchrys PTAL, thanks :)

File: r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java [hunk 1]
```diff
@@ -127,6 +128,8 @@ public final class MySqlConnectionConfiguration {
     @Nullable
     private final Publisher<String> passwordPublisher;
 
+    private final AddressResolverGroup<?> resolver;
```
Comment (jchrys) [inline r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java:132]: This field appears to be @Nullable. Additionally, there is no public method available to set this value. We need to support setting the resolver using Option methods.

You can refer to the following changes to add the new property and create test cases:
https://github.com/asyncer-io/r2dbc-mysql/pull/157

Please feel free to leave a comment if you need any further assistance.

**Comment by hezhangjian** [comment]
@jchrys Thank you for the reminder. I have fixed. However, I am still undecided on whether to name it resolver or resolverGroup. Do you have any preference?

File: r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java [hunk 2]
```diff
@@ -292,6 +297,10 @@ LoopResources getLoopResources() {
         return loopResources;
     }
 
+    AddressResolverGroup<?> getResolver() {
```
Comment (jchrys) [inline r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java]: `@Nullable` seems missing

**Comment by jchrys** [comment]
> @jchrys Thank you for the reminder. I have fixed. However, I am still undecided on whether to name it resolver or resolverGroup. Do you have any preference?

 I prefer "resolverGroup", since it refers to the underlying framework, keeping it as "resolverGroup" avoids any potential confusion.

**Comment by jchrys** [comment]
> @jchrys Thank you for the reminder. I have fixed. However, I am still undecided on whether to name it resolver or resolverGroup. Do you have any preference?

Consider Reactor netty refers to it as `resolver`, I think sticking with resolver would be fine. :D

File: r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java (hunk 1 above)
Comment (jchrys) [inline r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java:132]:
```suggestion
    @Nullable
    private final AddressResolverGroup<?> resolver;
```

File: r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java [hunk 3]
```diff
@@ -1156,6 +1169,11 @@ public Builder passwordPublisher(Publisher<String> passwordPublisher) {
             return this;
         }
 
+        public Builder resolver(AddressResolverGroup<?> resolver) {
```
Comment (jchrys) [inline r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java:1173]: Could you please add the JavaDoc?

**Review by jchrys — CHANGES_REQUESTED** [review]
Just one last detail: I believe we have one final step to complete. Could you please check the following?
`MySqlConnectionConfiguration#equals`
`MySqlConnectionConfiguration#hashCode`
`MySqlConnectionConfiguration#toString`

`MySqlConnectionConfiguration#setup(ConnectionFactoryOptions)`
Thank you!

File: r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java (hunk 3 above)
Comment (hezhangjian) [reply r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java:1173]: @jchrys Fixed

**Comment by hezhangjian** [comment]
@jchrys Thanks for your effort. PTAL again

File: r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/client/Client.java [hunk 4]
```diff
@@ -132,7 +133,7 @@ public interface Client {
      */
     static Mono<Client> connect(MySqlSslConfiguration ssl, SocketAddress address, boolean tcpKeepAlive,
         boolean tcpNoDelay, ConnectionContext context, @Nullable Duration connectTimeout,
-        LoopResources loopResources) {
+        LoopResources loopResources, @Nullable AddressResolverGroup<?> resolverGroup) {
```
Comment (jchrys) [inline r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/client/Client.java]: nit :D
```suggestion
        LoopResources loopResources, @Nullable AddressResolverGroup<?> resolver) {
```

File: r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/client/Client.java [hunk 5]
```diff
@@ -150,6 +151,10 @@ static Mono<Client> connect(MySqlSslConfiguration ssl, SocketAddress address, bo
             tcpClient = tcpClient.option(ChannelOption.TCP_NODELAY, tcpNoDelay);
         }
 
+        if (resolverGroup != null) {
+            tcpClient = tcpClient.resolver(resolverGroup);
```
Comment (jchrys) [inline r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/client/Client.java]:
```suggestion
            tcpClient = tcpClient.resolver(resolver);
```

File: r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/client/Client.java [hunk 6]
```diff
@@ -150,6 +151,10 @@ static Mono<Client> connect(MySqlSslConfiguration ssl, SocketAddress address, bo
             tcpClient = tcpClient.option(ChannelOption.TCP_NODELAY, tcpNoDelay);
         }
 
+        if (resolverGroup != null) {
```
Comment (jchrys) [inline r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/client/Client.java]:
```suggestion
        if (resolver != null) {
```

File: r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionFactoryProvider.java [hunk 7]
```diff
@@ -389,6 +401,7 @@ static MySqlConnectionConfiguration setup(ConnectionFactoryOptions options) {
             .to(builder::loopResources);
         mapper.optional(PASSWORD_PUBLISHER).as(Publisher.class)
             .to(builder::passwordPublisher);
+        mapper.optional(RESOLVER).as(AddressResolverGroup.class);
```
Comment (jchrys) [inline r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionFactoryProvider.java]:
```suggestion
        mapper.optional(RESOLVER).as(AddressResolverGroup.class)
            .to(builder::resolver);
```

File: r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java [hunk 8]
```diff
@@ -352,52 +364,39 @@ public int hashCode() {
...
     }
 
     @Override
     public String toString() {
-        if (isHost) {
-            return "MySqlConnectionConfiguration{host='" + domain + "', port=" + port + ", ssl=" + ssl +
-                ", tcpNoDelay=" + tcpNoDelay + ", tcpKeepAlive=" + tcpKeepAlive +
-                ", connectTimeout=" + connectTimeout +
+        return "MySqlConnectionConfiguration{" +
+                (isHost ? "host='" + domain + "', port=" + port + ", ssl=" + ssl : "unixSocket='" + domain + "'") +
```
Comment (jchrys) [inline r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlConnectionConfiguration.java]:
```suggestion
                (isHost ? "host='" + domain + "', port=" + port + ", ssl=" + ssl + 
                          ", tcpNoDelay=" + tcpNoDelay + ", tcpKeepAlive=" + tcpKeepAlive : 
                        "unixSocket='" + domain + "'") +
```

**Review by jchrys — APPROVED** [review]
@shoothzj 
Thanks a lot! :D

**Pull request merged**